src/hotspot/share/services/mallocTracker.hpp

```diff
@@ -150,14 +150,20 @@ class MallocMemorySnapshot : public ResourceObj {
     return &_malloc[index];
   }
 
-  inline MemoryCounter* malloc_overhead() {
-    return &_tracking_header;
+  inline size_t malloc_overhead() const {
+    return _all_mallocs.count() * sizeof(MallocHeader);
   }
 
   // Total malloc invocation count
-  size_t total_count() const;
+  size_t total_count() const {
+    return _all_mallocs.count();
```

Comment:
Is this correct style to implement the method in the header? Shouldn't it maybe be `inline`?

Reply:
I think it is okay for oneliners. And "inline" is implied if the definition appears in the class body.